src/components/searchBar/SearchBar.tsx

```diff
@@ -0,0 +1,54 @@
+import { FC, useContext, useEffect, useState } from "react";
+import { ThemeContext } from "../../ThemeContext";
+import "./SearchBar.css";
+
+type Props = {
+  searchTerm: string;
+  setSearchTerm: (searchTerm: string) => void;
+};
+
+const SearchBar: FC<Props> = ({ searchTerm, setSearchTerm }) => {
+  const [error, setError] = useState<Error | null>(null);
+  const [inputTerm, setInputTerm] = useState<string>(searchTerm);
+  const theme = useContext(ThemeContext);
+
+  const handleSearch = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setSearchTerm(inputTerm);
+  };
+
+  const throwError = () => {
+    console.log("Throwing error");
+    setError(new Error("Simulated error."));
+  };
+
+  useEffect(() => {
+    if (error) {
+      throw new Error("Simulated error.");
+    }
+  }, [error]);
+
+  return (
+    <form onSubmit={handleSearch} className="search-bar">
+      <input
+        className={
+          theme === "light"
+            ? "search-input light-search-input"
+            : "search-input dark-search-input"
```

Comment:
```suggestion
  "search-input ${theme}-search-input"
```
ну ты поняла. дальше не буду уже править. и переходи на модули)

там это чуть иначе можно хендлить